tests/PyPI-pytest/run_all.py

```diff
@@ -0,0 +1,211 @@
+#!/usr/bin/env python
+#     Copyright 2019, Kay Hayen, mailto:kay.hayen@gmail.com
+#
+#     Python test originally created or extracted from other peoples work. The
+#     parts from me are licensed as below. It is at least Free Software where
+#     it's copied from other people. In these cases, that will normally be
+#     indicated.
+#
+#     Licensed under the Apache License, Version 2.0 (the "License");
+#     you may not use this file except in compliance with the License.
+#     You may obtain a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#     Unless required by applicable law or agreed to in writing, software
+#     distributed under the License is distributed on an "AS IS" BASIS,
+#     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#     See the License for the specific language governing permissions and
+#     limitations under the License.
+#
+
+import os
+import sys
+import subprocess
+
+from nuitka.tools.testing.OutputComparison import compareOutput
+from nuitka.tools.testing.Virtualenv import withVirtualenv
+from nuitka.utils.FileOperations import removeDirectory
+
+
+# currently trying to start automating with urllib3, 
+#   will extend to other PyPI packages in the future
+
+packages = {
+    "urllib3" : {
+        "url": "https://github.com/urllib3/urllib3.git",
+    }
+}
+
+base_dir = os.getcwd()
+
+
+for package_name, details in packages.items():
+    try:
+        # get normal pytest results
+        with withVirtualenv("venv_uncompiled_%s" % package_name) as venv:
+            
+            venv.runCommand(
+                commands=[
+                    "git clone %s" % details["url"],
+                    "git clone https://github.com/nuitka/nuitka.git",
+                    "cd nuitka",
+                    "python setup.py develop",
+                    "cd ../%s" % package_name,
+                    "python -m pip install -r dev-requirements.txt",
+                    "python setup.py bdist_wheel",
+                ]
+            )
+            
+
+            dist_dir = os.path.join(venv.getVirtualenvDir(), package_name, "dist")
+
+            uncompiled_stdout, uncompiled_stderr = venv.runCommand(
```

Comment:
`runCommand` is called from here. I tried to change up runCommand to use subprocess and have it return stdout,stderr but my changes seemed to have no effect. I then commented out the entire `runCommand` method and for some reason, this file still ran correctly?

Reply:
From what I can see, the `withVirtualenv` context manager yields an object of the `Virtualenv` class, and the `runCommand` method is being called here. So how come my changes don't seem to take effect?